BullsAndCows/BullsAndCowsGame.cs

```diff
@@ -1,20 +1,48 @@
 ﻿using System;
+using System.Linq;
 
 namespace BullsAndCows
 {
     public class BullsAndCowsGame
     {
         private readonly SecretGenerator secretGenerator;
+        private readonly string secret = string.Empty;
         public BullsAndCowsGame(SecretGenerator secretGenerator)
         {
             this.secretGenerator = secretGenerator;
+            this.secret = this.secretGenerator.GenerateSecret();
         }
 
         public bool CanContinue => true;
 
         public string Guess(string guess)
         {
-            throw new NotImplementedException();
+            var guessWithoutSpace = guess.Replace(" ", string.Empty);
+            return this.Compare(this.secret, guessWithoutSpace);
+        }
+
+        public bool IsValid(string guess)
+        {
+            guess = guess.TrimEnd();
+            var guessWithoutSpace = guess.Replace(" ", string.Empty);
+            string validNumbers = "0123456789";
+            int countDuplicate = guessWithoutSpace.Where(inputChar => guessWithoutSpace.Replace(inputChar.ToString(), string.Empty).Contains(inputChar)).ToList().Count();
```

Comment:
Bug: when you replace inputchar with empty string, you are of course without duplication any more
Example: 1 2 3 4 2
inputchar = 2
after replace: 134
contains? false!

Reply:
Bug: if guess = "12 3 4"

Reply:
wrong logic, all duplicated char will be replaced by empty string.

Reply:
> Bug: if guess = "12 3 4"

I check the input length with "guess.Length != 7", I should name "7" to "inputLength"

Reply:
> Bug: if guess = "12 3 4"

if "12 3 4 " then?